library/scripts/library.mk

```diff
@@ -91,7 +91,7 @@ xilinx: component.xml
 component.xml: $(XILINX_DEPS) $(_XILINX_INTF_DEPS) $(_XILINX_LIB_DEPS)
 	-rm -rf $(CLEAN_TARGET)
 	$(call build, \
-		$(VIVADO) $(LIBRARY_NAME)_ip.tcl, \
+		flock component.xml -c "$(VIVADO) $(LIBRARY_NAME)_ip.tcl", \
```

Comment:
Thanks you for your initiative and effort. This optimization will be appreciated by everyone.

Let me continue  with what I would change. There are two issues at this line:
1. flock creates an empty component.xml, regardless of the success/fail status, an component.xml will result after the make call. So, when someone will call make, for the second time, a library that failed, will not be rebuild. Because the component.xml is there.
2. We want any error in `the library build to stop the whole build. You wrote this in another commit:
"Implicitly, this also fixes missing build dependencies such as:
	make: *** No rule to make target '../../../library/axi_clkgen/component.xml', needed by 'adrv9009_zcu102.sdk/system_top.xsa'.  Stop."
This is not a feature for us. If someone uses a release or stable branch, it does not matter that much. But in development it surely does. If a library build fails, the project build that depends on that library will fail in the end.
With this in mind the uglier scenarios will arise on the support forums. For us, trying to figure out why a project build fails for someone who does not give us the whole log or does not states the modifications of libraries in a reference design.
Also, working on a project with high resource demands, for which a build takes 3 hours or more will result in wasted time and resources for a developer or for our CI flow.
My suggestions is  to remove the component.xml if the build fails and pass the error status 

        $(call build, \
                flock component.xml -c "$(VIVADO) $(LIBRARY_NAME)_ip.tcl", \
                $(LIBRARY_NAME)_ip.log, \
                $(HL)$(LIBRARY_NAME)$(NC) library) || (ERR=$$? && rm component.xml && exit $$ERR);

If you have a better solution, please add  it.

Reply:
Hi @AndreiGrozav,

Thanks for testing this, and apologies for the delay. 

> 1. flock creates an empty component.xml, regardless of the success/fail status, an component.xml will result after the make call. So, when someone will call make, for the second time, a library that failed, will not be rebuild. Because the component.xml is there.

interesting! I definitely missed that. I like your approach, I can't really find anything within `flock` to take care of cleaning up if the command fails. 

actually, maybe [`.DELETE_ON_ERROR`](https://www.gnu.org/software/make/manual/make.html#index-_002eDELETE_005fON_005fERROR) would work here. I don't know if we want to be more conservative and take care of this in individual targets.

Do you have a preference? 

> 2. We want any error in `the library build to stop the whole build.

I definitely agree with that, the commit message might be misleading. If I'm not mistaken, this error message happens if you try using `make -j` before applying any commits from this branch.
Assuming I address the issue in 1. if an IP core build fails the project target won't get run because `make` will detect an issue in build dependencies. The only difference with master is that it would finish current builds.